preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp

```diff
@@ -25,22 +29,33 @@ class UI
 {
 public:
     HRESULT ShowUI();
+	HRESULT LaunchInstalledApp();
 
     static HRESULT Make(_In_ MsixRequest* msixRequest, _Out_ UI** instance);
     ~UI() {}
 private:
     MsixRequest* m_msixRequest = nullptr;
 
     HANDLE m_buttonClickedEvent;
+    HANDLE m_launchAppEvent;
 
     UI() {}
-    UI(_In_ MsixRequest* msixRequest) : m_msixRequest(msixRequest) { m_buttonClickedEvent = CreateEvent(NULL, FALSE, FALSE, NULL); }
+    UI(_In_ MsixRequest* msixRequest) : m_msixRequest(msixRequest) 
+	{
+		m_buttonClickedEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+		m_launchAppEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+	}
     
 public:
     HRESULT DisplayPackageInfo(HWND hWnd, RECT windowRect, std::wstring& displayText, std::wstring& messageText);
     int CreateInitWindow(HINSTANCE hInstance, int nCmdShow, const std::wstring& windowClass, const std::wstring& title);
 
     void SetButtonClicked() { SetEvent(m_buttonClickedEvent); }
+    void SetLaunchButtonClicked() { SetEvent(m_launchAppEvent); }

```

Comment:
>SetLaunchButtonClicked [](start = 9, length = 22)

i don't see this being called. did i miss it or is there a future task to call this?

Reply:
I created the event so the UI will wait without exiting while waiting for the launch button to be clicked at the same time using MsgWaitForMultipleObjects. The event is not being set anywhere.

Reply:
how does it work currently if you aren't setting the event anywhere? i guess what i was asking was did you mean to call it somewhere and forget to add it? or are you planning on calling it somewhere in the future checkin as part of a different task? or are you finished with this and intend for it never to be set?

---
In reply to: [264356252](https://github.com/Microsoft/msix-packaging/pull/59#discussion_r264356252) [](ancestors = 264356252)